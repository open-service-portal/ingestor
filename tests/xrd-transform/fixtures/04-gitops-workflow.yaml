apiVersion: apiextensions.crossplane.io/v2
kind: CompositeResourceDefinition
metadata:
  name: applications.platform.com
  annotations:
    backstage.io/description: "Application with GitOps PR workflow"
    backstage.io/techdocs-ref: dir:../../docs
    openportal.dev/template-steps: "gitops"
    openportal.dev/template-parameters: "gitops"
    openportal.dev/tags: "application,deployment"
spec:
  group: platform.com
  names:
    kind: Application
    plural: applications
  scope: Namespaced
  versions:
    - name: v1alpha1
      served: true
      referenceable: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              properties:
                image:
                  type: string
                  description: Container image
                replicas:
                  type: integer
                  description: Number of replicas
                  minimum: 1
                  maximum: 10
                  default: 1
                port:
                  type: integer
                  description: Service port
                  default: 8080
              required:
                - image
