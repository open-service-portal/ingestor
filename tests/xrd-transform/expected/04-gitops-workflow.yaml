apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: applications-platform-com
  title: Application
  description: Create and manage Application resources
  tags:
    - crossplane
    - platform-com
    - application
    - deployment
  annotations:
    backstage.io/managed-by: xrd-transform
    backstage.io/managed-by-location: provider:XRDTemplateEntityProvider
    backstage.io/managed-by-origin-location: provider:XRDTemplateEntityProvider
    crossplane.io/xrd-name: applications.platform.com
    crossplane.io/xrd-group: platform.com
    backstage.io/techdocs-ref: dir:../../docs
spec:
  owner: platform-team
  type: crossplane-resource
  parameters:
    - title: Resource Configuration
      required:
        - name
        - namespace
        - image
      properties:
        name:
          title: Name
          type: string
          description: Name of the Application
          pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
        namespace:
          title: Namespace
          type: string
          description: Kubernetes namespace
          default: default
        image:
          title: Image
          type: string
          description: Container image
        replicas:
          title: Replicas
          type: number
          description: Number of replicas
          default: 1
          minimum: 1
          maximum: 10
        port:
          title: Port
          type: number
          description: Service port
          default: 8080
  steps:
    - id: workflow-info
      name: Direct Apply Workflow Information
      action: debug:log
      input:
        message: |
          ‚ö° This template uses direct apply workflow:

          1. Creates your Application resource immediately in the cluster
          2. Registers the resource in the Backstage catalog

          Your resource will be created as soon as you click Create.
      output:
        text:
          - title: ‚ö° Direct Apply Workflow
            content: >
              This template follows a **direct apply workflow** for immediate
              resource creation:


              **What happens when you click Create:**


              1. **Create Resource Immediately** üöÄ
                 - Your Application resource is created directly in the Kubernetes cluster
                 - No approval required - resource is active immediately
                 - Applied to cluster: `rancher-desktop`
                 - Created in namespace: `$`

              2. **Register in Catalog** üìù
                 - The resource is automatically registered in the Backstage software catalog
                 - You can track and manage it through Backstage

              **‚úÖ Important**: Your resource is **created immediately** without
              approval workflow.
    - id: create-resource
      name: Create Application
      action: kube:apply
      input:
        manifest: |
          apiVersion: platform.com/v1alpha1
          kind: Application
          metadata:
            name: ${{ parameters.name }}
            namespace: ${{ parameters.namespace }}
          spec: ${{ parameters }}
        namespaced: true
      output:
        text:
          - title: Resource Details
            content: |
              **Resource Created Successfully!** ‚úÖ

              **Resource Information:**
              - **Name**: `$`
              - **Type**: Application
              - **API Version**: platform.com/v1alpha1
              - **Cluster**: rancher-desktop
              - **Namespace**: `$`

              Your resource is now active in the cluster and ready to use.
    - id: register
      name: Register in Catalog
      action: catalog:register
      input:
        repoContentsUrl: ${{ steps.create-resource.output.repoContentsUrl }}
        catalogInfoPath: /catalog-info.yaml
      output:
        links:
          - title: View in Catalog
            icon: catalog
            url: ${{ steps['register'].output.entityRef }}
        text:
          - title: Catalog Registration
            content: |
              **Registered in Backstage Catalog** üìù

              Your Application resource has been added to the software catalog.
              You can now track, monitor, and manage it through Backstage.

